crawl-ref/source/l-crawl.cc

```diff
@@ -1781,6 +1781,12 @@ LUAWRAP(crawl_clear_message_store, clear_message_store())
  */
 LUARET1(crawl_seen_hups, number, crawl_state.seen_hups)
 
+LUAFN(crawl_setfenv)
+{
+    lua_setupvalue(ls, 1, 1);
```

Comment:
Having taken a quick glance (and also in the past spent a bunch of time googling the issue), it's not entirely obvious to me how/if this is a full equivalent to setfenv; could you document this at all? (Was setfenv really doing nothing more than setting the upvalues? Most of the code I saw for trying to emulate setfenv was more involved, though a lot of what I found was also on the lua side. This is a fairly crucial issue as clua sandboxing relies on setfenv...)